src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -169,4 +179,169 @@ public static String composeClassHeaderEndpointSampleCode(
+        composeUnaryRpcMethodSampleCode(method, arguments, clientType, resourceNames));
+  }
+
+  public static TryCatchStatement composeUnaryRpcMethodSampleCode(
+      Method method,
+      List<MethodArgument> arguments,
+      TypeNode clientType,
+      Map<String, ResourceName> resourceNames) {
+    VariableExpr clientVarExpr =
+        VariableExpr.withVariable(
+            Variable.builder()
+                .setName(JavaStyle.toLowerCamelCase(clientType.reference().name()))
+                .setType(clientType)
+                .build());
+    // Assign each method arguments with its default value.
+    List<Expr> methodArgExprs = createMethodArgVarExprs(arguments, resourceNames);
+    List<Expr> bodyExpr =
+        assignMethodArgumentsWithDefaultValues(arguments, methodArgExprs, resourceNames);
+    // Invoke current method based on return type.
+    // e.g. if return void, echoClient.echo(..); or,
+    // e.g. if return other type, EchoResponse response = echoClient.echo(...);
+    boolean returnsVoid = isProtoEmptyType(method.outputType());
+    if (returnsVoid) {
+      bodyExpr.add(
+          MethodInvocationExpr.builder()
+              .setExprReferenceExpr(clientVarExpr)
+              .setMethodName(JavaStyle.toLowerCamelCase(method.name()))
+              .setArguments(methodArgExprs)
+              .setReturnType(clientType)
+              .build());
+    } else {
+      VariableExpr responseVarExpr =
+          VariableExpr.withVariable(
+              Variable.builder().setName(RESPONSE).setType(method.outputType()).build());
+      bodyExpr.add(
+          createAssignExprForVariableWithClientMethod(
+              responseVarExpr,
+              clientVarExpr,
+              JavaStyle.toLowerCamelCase(method.name()),
+              methodArgExprs));
+    }
+
+    return TryCatchStatement.builder()
+        .setTryResourceExpr(assignClientVariableWithCreateMethodExpr(clientVarExpr))
+        .setTryBody(
+            bodyExpr.stream().map(e -> ExprStatement.withExpr(e)).collect(Collectors.toList()))
+        .setIsSampleCode(true)
+        .build();
+  }
+
+  // ==================================Helpers===================================================//
+
+  // Assign client variable expr with create client.
+  // e.g EchoClient echoClient = EchoClient.create()
+  private static AssignmentExpr assignClientVariableWithCreateMethodExpr(
+      VariableExpr clientVarExpr) {
+    return AssignmentExpr.builder()
+        .setVariableExpr(clientVarExpr.toBuilder().setIsDecl(true).build())
+        .setValueExpr(
+            MethodInvocationExpr.builder()
+                .setStaticReferenceType(clientVarExpr.variable().type())
+                .setReturnType(clientVarExpr.variable().type())
+                .setMethodName("create")
+                .build())
+        .build();
+  }
+
+  // Create a list of Expr for method argument variable expression.
+  private static List<Expr> createMethodArgVarExprs(
```

Comment:
Can we rename this method to more specifically represent its summarized behavior? E.g. createClientMethodArgVarExprs or createRpcMethodArgVarExprs.